domain/secret/state/state.go

```diff
@@ -2900,6 +2870,39 @@ AND    (subject_type_id = $secretAccessorType.unit_type_id AND subject_id IN ($u
 	return revisionResult, nil
 }
 
+// GetSecretRevisionID returns the revision UUID for the specified secret URI and version,
+// or an error satisfying [secreterrors.SecretRevisionNotFound] if the revision is not found.
+func (s *State) GetSecretRevisionID(ctx context.Context, uri *coresecrets.URI, revision int) (string, error) {
+	db, err := s.DB()
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+
+	input := secretRevision{
```

Comment:
This should really be called `secretRev` or something, not `input`. We don't use `input` anywhere else and it's quite meaningless. And worse, down below we do this

`tx.Query(ctx, stmt, input).Get(&input)`

Clearly, it's not input as we are selecting a result into it. So using `secretRev` like we do in other domain state packages is more clear and correct.